src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -17,23 +16,25 @@ public class MachineServiceImpl implements MachineService<Machine> {
     private final ExcavatorProducer excavatorProducer = new ExcavatorProducer();
 
     @Override
-    public List<Machine> getAll(Class type) {
-        List<? extends Machine> machines = null;
-        if (type == Bulldozer.class) {
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = List.of();
+        if (Bulldozer.class.isAssignableFrom(type)) {
```

Comment:
I still don’t understand why we need `isAssignableFrom` here and how it is better than the usual `equals`((

Reply:
as I mentioned earlier we indeed can get around with just `==` or `equals` - actually they do the same, since `equals()` in `Class` class (I know sounds weird) isn't overriden from `Object` class

in this particular scenario the difference is next: 

suppose we have `Buldozer` and `BuldozerChild extends Buldozer`:

- if type = `Buldozer` - `isAssignableFrom()` will return true and `==` will return true
- if type = `BuldozerChild` - `isAssignableFrom()` will return true but `==` will return false

hopefully now you understand the difference

and again - usage of `isAssignableFrom()` is not mandatory here, since we care about exact classes match, my previous comment may be considered unnecessary :)